test/test_modelselection.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+from pypesto.modelselection.row2problem import *
+from pypesto.petab import PetabImporter
+import petab
+import pandas as pd
+
+
+@pytest.fixture
+def modelselection_file_example():
+    return "doc/example/model_selection/modelSelectionSpecification_example_" \
+           "modelSelection.tsv"
+
+
+@pytest.fixture
+def yaml_file_example():
+    return "doc/example/model_selection/example_modelSelection.yaml"
+
+
+@pytest.mark.xfail  # example does not seem to compile properly yet
+def test_row2problem_yaml_string(yaml_file_example):
+    petab_problem = petab.Problem.from_yaml(yaml_file_example)
+
+    importer = PetabImporter(petab_problem)
+    obj = importer.create_objective()
+    pypesto_problem = importer.create_problem(obj)
+
+    row = pd.Series()
+    assert row2problem(row, yaml_file_example) == pypesto_problem
+
```

Comment:
explicitly assert that `x_names`, `x_fixed_vals`, `x_fixed_ids` etc. are as expected.